internal/controller/progressive/progressive.go

```diff
@@ -404,13 +404,21 @@ func AssessUpgradingPipelineType(
 		WithValues("namespace", existingUpgradingChildDef.GetNamespace(), "name", existingUpgradingChildDef.GetName()).
 		Debugf("Upgrading child is in phase %s, conditions healthy=%t, ready replicas match desired replicas=%t", upgradingObjectStatus.Phase, healthyConditions, healthyReplicas)
 
-	if analysisStatus != nil {
-		if upgradingObjectStatus.Phase == "Running" && healthyConditions && healthyReplicas && analysisStatus.Phase == argorolloutsv1.AnalysisPhaseSuccessful {
-			return apiv1.AssessmentResultSuccess, "", nil
-		}
-	}
-
+	// conduct standard health assessment first
 	if upgradingObjectStatus.Phase == "Running" && healthyConditions && healthyReplicas {
+		// if analysisStatus is set with an AnalysisRun's name, we must also check that it is in a Completed phase to declare success
+		if analysisStatus != nil && analysisStatus.AnalysisRunName != "" {
+			switch analysisStatus.Phase {
+			case argorolloutsv1.AnalysisPhaseSuccessful:
+				return apiv1.AssessmentResultSuccess, "", nil
+			case argorolloutsv1.AnalysisPhaseError, argorolloutsv1.AnalysisPhaseFailed, argorolloutsv1.AnalysisPhaseInconclusive:
+				return apiv1.AssessmentResultFailure, "", nil
+			default:
+				numaLogger.WithValues("namespace", existingUpgradingChildDef.GetNamespace(), "name", existingUpgradingChildDef.GetName()).
```

Comment:
so, you're only going to log it if it's not error or success?

How about we moving the logging prior to the switch statement?

Reply:
From previous review, I had thought it was only for non complete phases. Will change that now.